src/System.Private.CoreLib/src/System/Threading/Thread.CoreRT.cs

```diff
@@ -169,14 +169,20 @@ public bool IsBackground
                 {
                     throw new ThreadStateException(SR.ThreadState_Dead_State);
                 }
-                // TODO: Keep a counter of running foregroung threads so we can wait on process exit
+                bool isBackground = GetThreadStateBit(ThreadState.Background);
```

Comment:
This could be called from multiple threads in paralell. Instead of checking upfront, SetThreadStateBit should return the previous state of the bit and then increment/decrement if there was a change.